tests/sync/sync_manager.cpp

```diff
@@ -85,34 +85,34 @@ TEST_CASE("sync_manager: `path_for_realm` API", "[sync]") {
 
     SECTION("should work properly without metadata") {
         TestSyncManager init_sync_manager({.base_path = base_path, .metadata_mode = SyncManager::MetadataMode::NoMetadata});
-        const auto expected = base_path + "mongodb-realm/app_id/foobarbaz/realms%3A%2F%2Frealm.example.org%2Fa%2Fb%2F%7E%2F123456%2Fxyz.realm";
+        const auto expected = base_path + "app_id/mongodb-realm/foobarbaz/realms%3A%2F%2Frealm.example.org%2Fa%2Fb%2F%7E%2F123456%2Fxyz.realm";
```

Comment:
I'm a bit uncomfortable doing it like this with the app_id as a prefix. At least on Android files are being stored in a shared files folder, which means we will potentially create many folders that might conflict with already existing files or folders. I would much prefer if we only had `mongodb-realm` and then something like `mongodb-realm/<appId>/server-utility` and `mongodb-realm/<appId>/recovered-realms`